onnxruntime/core/optimizer/matmul_scale_fusion.cc

```diff
@@ -15,6 +15,20 @@
 namespace onnxruntime {
 
 namespace {
+
+// TransposeScaleMatMul supports limited data types.
+static std::vector<std::string> supported_data_types{"tensor(float16)", "tensor(float)", "tensor(double)", "tensor(bfloat16)"};
```

Comment:
On the CPU side, TransposeScaleMatMul is only supported for float. Is there a way to check the EP's registry to see if it has a TransposeScaleMatMul that is compatible with the type?